engine/src/main/java/org/terasology/registry/CoreRegistry.java

```diff
@@ -69,4 +71,46 @@ public static void setContext(Context context) {
         return context.get(type);
     }
 
+    /**
+     * Returns this class as a Context object.
+     *
+     * @return A Context object representing this class.
+     */
+    public static Context asContext() {
+        return CoreRegistryContext.INSTANCE;
```

Comment:
As suggested above, I recommend that the thing you register at the context should have the same live span as the context. Thus using something that lives over multiple context lifespans does not make sense.